main.py

```diff
@@ -365,19 +375,33 @@ def compute_precond(x):
             buffer_nl_ff = np.array(buffer_nbrs_ff.idx)
 
             #Solve the saddle point problem in the positive direction
-            saddle_x = solver(
+            output_precompute = utils.precompute(buffer_positions, buffer_gaussian_grid_spacing, buffer_nl_ff, buffer_nl_lub, buffer_displacements_vector_matrix, xy,
+                                                 N, Lx, Ly, Lz, Nx, Ny, Nz,
+                                                 prefac, expfac, quadW,
+                                                 gaussP, gaussPd2,
+                                                 ewald_n, ewald_dr, ewald_cut, ewaldC1,
+                                                 ResTable_min, ResTable_dr, ResTable_dist, ResTable_vals)
+
+            saddle_x, exitcode_gmres = solver(
                 saddle_b,  # rhs vector of the linear system
                 gridk,
                 R_fu_prec_lower_triang,
-                utils.precompute(buffer_positions, buffer_gaussian_grid_spacing, buffer_nl_ff, buffer_nl_lub, buffer_displacements_vector_matrix, xy,
-                                 N, Lx, Ly, Lz, Nx, Ny, Nz,
-                                 prefac, expfac, quadW,
-                                 gaussP, gaussPd2,
-                                 ewald_n, ewald_dr, ewald_cut, ewaldC1,
-                                 ResTable_min, ResTable_dr, ResTable_dist, ResTable_vals)[::]
+                output_precompute
             )
+            if(exitcode_gmres > 0):
+                print('GMRES (RFD) did not converge! Iterations are ', exitcode_gmres)
+                raise ValueError("Abort!")
```

Comment:
I think it would be easier if the error raised has the error message you're currently printing. That way the error and the message are not separated.